core/templates/dev/head/pages/library/ActivityTilesInfinityGridDirective.js

```diff
@@ -20,14 +20,20 @@ oppia.directive('activityTilesInfinityGrid', [
   'UrlInterpolationService', function(UrlInterpolationService) {
     return {
       restrict: 'E',
+      scope: {
+        learnerDashboardActivityIds: '=learnerDashboardActivityIds',
+      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/library/' +
         'activity_tiles_infinity_grid_directive.html'),
       controller: [
-        '$scope', '$rootScope', 'searchService',
-        function($scope, $rootScope, searchService) {
+        '$scope', '$rootScope', '$http', 'searchService', 'alertsService',
+        '$modal', function(
+          $scope, $rootScope, $http, searchService, alertsService, $modal) {
           $scope.endOfPageIsReached = false;
           $scope.allActivitiesInOrder = [];
+          var hoverOverActivity = false;
```

Comment:
Seems like a verb. I'm not sure what this represents. Is it whether an activity is hovered over? If so, why not remove this and just use activeActivityId (or, perhaps, hoveredActivityId) to track this?